File: OOPs/Method Chaining/MethodChaining.java
class MethodChaining {
    private String brand;
    private String color;

    public MethodChaining setBrand(String brand) {
        this.brand = brand;
        return this;  // Return the current object
    }

    public MethodChaining setColor(String color) {
        this.color = color;
        return this;  // Return the current object
    }

    public void display() {
        System.out.println("MethodChaining: " + brand + ", Color: " + color);
    }

    public static void main(String[] args) {
        MethodChaining car = new MethodChaining();
        car.setBrand("Toyota").setColor("Red").display();
    }
}

File: OOPs/Method Chaining/method_chaining.cpp
#include <bits/stdc++.h>
using namespace std;

class Car {
    string brand;
    string color;

public:
    Car& setBrand(const string& b) {
        brand = b;
        return *this;  // Return the current object
    }

    Car& setColor(const string& c) {
        color = c;
        return *this;  // Return the current object
    }

    void display() const {
        cout << "Car: " << brand << ", Color: " << color << endl;
    }
};

int main() {
    Car car;
    car.setBrand("Toyota").setColor("Red").display();
    return 0;
}
